dev_guide/new_app.adoc

```diff
@@ -13,14 +13,15 @@ toc::[]
 Create a new OpenShift application by using the
 link:../architecture/infrastructure_components/web_console.html[web console] or
 by running the `oc new-app` command from the CLI. OpenShift creates a new
-application by specifying source code,
-link:../architecture/core_objects/openshift_model.html#template[templates], and
-link:../architecture/core_objects/openshift_model.html#image[images]. OpenShift
-looks for images on the local Docker installation (if available), in a
+application by specifying
+link:#specifying-source-code[source code],
+link:#specifying-an-image[images], or
+link:#specifying-a-template[templates]. The OpenShift
+new-app CLI command looks for images on the local Docker installation (if available), in a
 link:../architecture/infrastructure_components/image_registry.html[Docker
 registry], or an OpenShift image stream.
 
-If you specify a source code URL, OpenShift takes your source code
+If you specify source code, OpenShift takes your source code
```

Comment:
How about sometihng like: 
If you specify source code, new-app will attempt to construct a BuildConfig that will build your source into a new application image.  It will also construct a DeploymentConfig that will deploy that new image and a Service to provide load balanced access to the deployment that is running your image.